Pagination for Users + Users/UserID/Tasks Environ fix

kieranmiguel opened the pull request:
Allows for pagination variables to adjust the count and page of users that is selected. 
Format: /users/?count={count}&page={page}

When a request is made, the API "Divides" up the tasks based on the count of users that is selected. For example, is one queries for 10 count and page 5, and there are 50 tasks, tasks 40-50 will appear. If one queries for count 4 and page 5, tasks 16-20 will appear. 

Redis Caching works by setting an individual key for each pagination and count variable. So even queries that contain some of the same tasks will not pull from the same redis key. Redis caches are set to expire after a time of 1200 seconds. This means caching currently only supports identical requests. 

Testing:
To test, send a postman get request in this format: /users/?count=4&page=5 and make sure CACHE-TOGGLE = "On" in local.settings.json

The first request should query the cache and then return from the database,
a second subsequent request should query the cache. This information should be available in the terminal

kieranmiguel committed:
Environ Fix

kieranmiguel committed:
Merge branch 'development' of https://github.com/StopMakingUsernamesUnique/ad440-winter2021-thursday-repo into development

kieranmiguel committed:
Merge remote-tracking branch 'upstream/development' into development

kieranmiguel committed:
Url Query parse test

Prints debug message to test if count and page queries are working

farhadbahrehmandhenry commented on functions/HttpTriggerAPIUsers/__init__.py:
```diff
@@ -3,12 +3,21 @@
 def main(req: func.HttpRequest) -> func.HttpResponse:
+    url = urlparse.urlparse(req.url)
+    count = parse_qs(url.query)['count']
+    page = parse_qs(url.query)['page']
```
- where are you using count and page variables. 
- also, I think that would be a good idea to use an if statement to avoid crashing the app when count and page are not defined in url.

farhadbahrehmandhenry reviewed (commented):
HttpTriggerAPIUsersIdTask is not working at the moment. replacing line 126 with the following code should fix the problem.
json_tasks = json.dumps(tasks, default=default);

kieranmiguel committed:
Merge remote-tracking branch 'upstream/development' into development

kieranmiguel committed:
Pagination for User

When querying data, API now accepts page and count variables, and returns data based on these. Redis sever now includes keys for page and count variables

kieranmiguel committed:
Json Dumps Users/UserID/tasks Fix

Fixes json returning datetime values incorrectly

chonger878 reviewed (approved):
The documentation placed strategically on the code helps me understand what happens at each point of the program.  A slight suggestion for the task/PR page, perhaps separate the part about count 4 and page 5 or anything to do with queries, give it its' own paragraph, and name that as testing documentation or something...?

kieranmiguel committed:
Merge branch 'development' into development

gillilands19 commented:
Working for me. And it looks like you took care of the suggestions by Farhad.

gillilands19 reviewed (approved):
Working for me. And it looks like you took care of all the suggestions from Farhad.
